### DISCUSSION TITLE
Double-clicking a Raster/ToonzRaster/ToonzVector level in the FX/Stage Schematic to open its settings

### DISCUSSION THREAD

**Discussion by WolfInABowl** [opentoonz/opentoonz#6533] [Ideas]
As the title says, I believe it would make sense to open the `Level Settings` window when double-clicking a level in the FX/Stage Schematic windows.

**FX Schematic**
Double-clicking in the FX Schematic opens an empty FX Settings window. The only thing useful about this is the Swatch Viewer, though I do not believe this sees much use at all.
Double-clicking the `XSheet` and `Output` nodes opens the Scene Settings and Output Settings respectively, so it would make sense to apply this same behavior to levels as well.

**Stage Schematic**
Double clicking in the Stage Schematic does nothing, and unlike the FX Schematic, `Level Settings` does not appear as an option in the right-click context menu. This means that there is no way to access a level's settings within the Stage Schematic.
Double-clicking a camera opens its settings, so it would make sense to apply this same behavior to levels as well.

**Comment by gab3d** [comment]
Keep in mind what you see represented by a node in the Schematic is a Column/Layer, not a Level, and there could be several Levels exposed in a single Column/Layer. So, perhaps what should be shown on double clicking one of these nodes would be the column/layer Opacity level slider and the Filter dropdown menu options.

Still, I understand what you're after... I myself have proposed turning the current **FX Settings** panel into a general **Selected Item Settings** panel in the past, and display there the settings for the currently selected item in the interface, so everytime you single click on a node in the Schematic, a cell in the Xsheet/Timeline, a drawing frame in the Level Strip,  an item in the Scene Cast, a cell in the Function Editor spreadsheet, a Style in the Style Editor, etc. you could always find its options there.

The current FX Settings panel already has the ability to show different items settings in there via Tabs (currently only really used when you select a Macro FX, to show the settings of each FX node of the Macro), so perhaps that feature could be used for showing both Column settings and exposed Levels settings in the same panel, when you select a column node in the Schematic? 

I know that's not exactly what you were asking for, but considering that currently there's a lot of time ahead of the next release, perhaps it's the right time to think about a more global solution to that kind of UI needs.

**Comment by RodneyBaker** [comment]
There are good ideas and thoughts flowing in this request.
Recommend revisiting this in the v1.8 development cycle to determine what can be implemented.
For instance, adding Level Settings to the context sensitive menu for column nodes selected seems quite reasonable.

Aside:  
We should also look at the pros and cons of allowing users access to customize Right Click menus.
This might be difficult due to the context sensitivie nature of those menus but if we cannot grant easy access to modification at least we can document the process of how users can request changes to those menus and better understand how they can change them on their own via the source code when building Opentoonz locally (i.e. for maximum customization).

Workarounds:
One of the main reasons I'm adding this note it to mention that use of the Level Settings shortcut key does work in the Stage Schematic and the results of the level settings displayed will vary depending on whether a column node is selected or not.  If no column node is selected then I believe the first level encountered in the xsheet/timeline will have it's settings opened.  That can provide at least a cursory workaround. 

I do find the fact that double clicking on a node in stage schematic at present does nothing is a good hint that we could provide that basic level settings popup functionality.  A potential issue with consistency does exist in the FX Schematic though where it opens the empty FX node view with swatch viewer.  I'm not sure how we would best reconcile the difference between these two unless some deeper updates are made related to Level Settings and the FX node popup.
